src/gen/js/service.ts.template

```diff
@@ -3,8 +3,8 @@
 import spec from './spec';;
 
 export class ServiceError extends Error implements api.ServiceError {
-	status: number;;
-	code: number;;
+	status: number = -1;;
+	code: number = -1;;
```

Comment:
Just a quick note on this one. Changing these to `-1` defaults I'm hesitant to do as it would be classed as a breaking change so require a major bump. Also some people may genuinely have a `-1` error code from the server (I know unlikely).

Let me know your thinking though, perhaps theres a good case I'm missing.

Reply:
This can be fixed in a different way given that the class has a constructor that assigns these mandatory fields and the methods that construct the objects never leave any of them undefined, but it's either this or flagging status and code as optional. I personally detest optional fields so my preferred way would probably be to rewrite the type use... What thinkest thou?

Reply:
Was TS complaining about these not having a value assigned?

Reply:
Yes, as is openapi-client is not compatible with strict mode projects